tests/sanity/sanity_test.go

```diff
@@ -0,0 +1,304 @@
+	"github.com/kubernetes-sigs/aws-ebs-csi-driver/pkg/util"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials/insecure"
+)
+
+var (
+	disks            = make(map[string]*cloud.Disk)
+	snapshots        = make(map[string]*cloud.Snapshot)
+	snapshotNameToID = make(map[string]string)
+)
+
+func TestSanity(t *testing.T) {
+	defer func() {
+		if r := recover(); r != nil {
+			t.Errorf("Test panicked: %v", r)
+		}
+	}()
+
+	mockCtrl := gomock.NewController(t)
+	defer mockCtrl.Finish()
+
+	tmpDir := "/tmp/csi"
+	endpoint := fmt.Sprintf("unix:%s/csi.sock", tmpDir)
+	mountPath := path.Join(tmpDir, "mount")
+	stagePath := path.Join(tmpDir, "stage")
+
+	err := os.MkdirAll(tmpDir, 0755)
+	if err != nil {
+		t.Fatalf("Failed to create sanity temp working dir %s: %v", tmpDir, err.Error())
+	}
+
+	defer func() {
+		if err = os.RemoveAll(tmpDir); err != nil {
+			t.Fatalf("Failed to clean up sanity temp working dir %s: %v", tmpDir, err.Error())
+		}
+	}()
+
+	fakeMetaData := &cloud.Metadata{
+		InstanceID: "i-1234567890abcdef0",
+		Region:     "us-west-2",
+	}
+
+	config := csisanity.TestConfig{
+		TargetPath:           mountPath,
+		StagingPath:          stagePath,
+		CreateTargetDir:      createDir,
+		CreateStagingDir:     createDir,
+		Address:              endpoint,
+		DialOptions:          []grpc.DialOption{grpc.WithTransportCredentials(insecure.NewCredentials())},
+		IDGen:                csisanity.DefaultIDGenerator{},
+		TestVolumeSize:       10 * util.GiB,
+		TestVolumeAccessType: "mount",
+	}
+
+	fakeMounter := d.NewMockMounter(mockCtrl)
+	fakeMounter.EXPECT().Unpublish(gomock.Any()).Return(nil).AnyTimes()
+	fakeMounter.EXPECT().GetDeviceNameFromMount(gomock.Any()).Return("", 0, nil).AnyTimes()
+	fakeMounter.EXPECT().Unstage(gomock.Any()).Return(nil).AnyTimes()
+	fakeMounter.EXPECT().FormatAndMountSensitiveWithFormatOptions(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+	fakeMounter.EXPECT().NeedResize(gomock.Any(), gomock.Any()).Return(false, nil).AnyTimes()
+	fakeMounter.EXPECT().MakeDir(gomock.Any()).Return(nil).AnyTimes()
+	fakeMounter.EXPECT().IsLikelyNotMountPoint(gomock.Any()).Return(true, nil).AnyTimes()
+	fakeMounter.EXPECT().Mount(gomock.Any(), gomock.Any(), gomock.Any(), gomock.Any()).Return(nil).AnyTimes()
+
+	identifier := d.NewMockDeviceIdentifier(mockCtrl)
+	identifier.EXPECT().Lstat(gomock.Any()).Return(nil, nil).AnyTimes()
+
+	fakeMounter.EXPECT().PathExists(gomock.Any()).DoAndReturn(
+		func(targetPath string) (bool, error) {
+			if targetPath == mountPath {
+				return true, nil
+			}
+			return false, nil
+		},
+	).AnyTimes()
+
+	r := d.NewMockResizefs(mockCtrl)
+	fakeMounter.EXPECT().NewResizeFs().Return(r, nil).AnyTimes()
+	r.EXPECT().Resize(gomock.Any(), gomock.Any()).Return(true, nil).AnyTimes()
+
+	fakeCloud := cloud.NewMockCloud(mockCtrl)
+
+	// CreateDisk
+	fakeCloud.EXPECT().CreateDisk(gomock.Any(), gomock.Any(), gomock.Any()).DoAndReturn(
```

Comment:
Personal Nit: Extract this cloud mocking into function named `prepareMockCloud`.

Out of scope for PR: These `DoAndReturns` suggest that Cloud and Driver are too tightly coupled (Probably because Cloud package does more far more than just act as a wrapper for EC2 and EBS Apis.)

Maybe we can utilize [Facade](https://refactoring.guru/design-patterns/facade)-style wrappers for EC2  and EBS APIs, and a separate [Adapter](https://refactoring.guru/design-patterns/adapter) package that translates between AWS-land and CSI-land (This adapter could convert an EC2 error into a csi/grpc error).

Reply:
> Extract this cloud mocking into function named prepareMockCloud

Fixed ✅